src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -941,6 +946,49 @@ TEST_F(ApplicationManagerImplTest, UnregisterAnotherAppDuringAudioPassThru) {
   }
 }
 
+TEST_F(ApplicationManagerImplTest,
+       RegisterApplication_TrueIconPath_PathToTheIconWasSet) {
+  file_system::CreateDirectory(kDirectoryName);
+  const std::string full_icon_path = kDirectoryName + "/" + kAppId;
+  file_system::CreateFile(full_icon_path);
+
+  smart_objects::SmartObject request_for_registration(
+      smart_objects::SmartType_Map);
+
+  smart_objects::SmartObject& params =
+      request_for_registration[strings::msg_params];
+  params[strings::app_id] = kAppId;
+  params[strings::language_desired] = mobile_api::Language::EN_US;
+  params[strings::hmi_display_language_desired] = mobile_api::Language::EN_US;
+
+  request_for_registration[strings::params][strings::connection_key] =
+      kConnectionKey;
+  request_for_registration[strings::msg_params][strings::app_name] = kAppName;
+  request_for_registration[strings::msg_params][strings::sync_msg_version]
+                          [strings::minor_version] = APIVersion::kAPIV2;
+  request_for_registration[strings::msg_params][strings::sync_msg_version]
+                          [strings::major_version] = APIVersion::kAPIV3;
+
+  request_for_registration[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
+
+  smart_objects::SmartObjectSPtr request_for_registrationSPtr =
+      MakeShared<smart_objects::SmartObject>(request_for_registration);
+
+  ApplicationSharedPtr application =
+      app_manager_impl_->RegisterApplication(request_for_registrationSPtr);
+  EXPECT_STREQ(kAppName.c_str(), application->name().c_str());
```

Comment:
@BSolonenko Compare without redundant conversation

Reply:
@AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2121/commits/d3669007e038ee0921f3d6fa4b3786b723fed263